packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -0,0 +1,354 @@
+
+      const { blockhash } = await this.connection.getLatestBlockhash()
+
+      const computeUnitLimit = chainSpecific.computeUnitLimit
+        ? Number(chainSpecific.computeUnitLimit)
+        : undefined
+
+      const computeUnitPrice = chainSpecific.computeUnitPrice
+        ? Number(chainSpecific.computeUnitPrice)
+        : undefined
+
+      const txToSign: SignTx<KnownChainIds.SolanaMainnet> = {
+        addressNList: toAddressNList(this.getBIP44Params({ accountNumber })),
+        blockHash: blockhash,
+        computeUnitLimit,
+        computeUnitPrice,
+        // TODO: handle extra instructions
+        instructions: undefined,
+        to,
+        value,
+      }
+
+      return { txToSign }
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async signTransaction(signTxInput: SignTxInput<SolanaSignTx>): Promise<string> {
+    try {
+      const { txToSign, wallet } = signTxInput
+
+      if (!supportsSolana(wallet)) throw new Error('Wallet does not support Solana.')
+
+      const signedTx = await wallet.solanaSignTx(txToSign)
+
+      if (!signedTx?.serialized) throw new Error('Error signing tx')
+
+      return signedTx.serialized
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async signAndBroadcastTransaction({
+    senderAddress,
+    receiverAddress,
+    signTxInput,
+  }: SignAndBroadcastTransactionInput<KnownChainIds.SolanaMainnet>): Promise<string> {
+    try {
+      const { txToSign, wallet } = signTxInput
+
+      await Promise.all([
+        assertAddressNotSanctioned(senderAddress),
+        receiverAddress !== CONTRACT_INTERACTION && assertAddressNotSanctioned(receiverAddress),
+      ])
+
+      if (!supportsSolana(wallet)) throw new Error('Wallet does not support Solana.')
+
+      const tx = await wallet.solanaSendTx?.(txToSign)
+
+      if (!tx) throw new Error('Error signing & broadcasting tx')
+
+      return tx.signature
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async broadcastTransaction({
+    senderAddress,
+    receiverAddress,
+    hex,
+  }: BroadcastTransactionInput): Promise<string> {
+    try {
+      await Promise.all([
+        assertAddressNotSanctioned(senderAddress),
+        receiverAddress !== CONTRACT_INTERACTION && assertAddressNotSanctioned(receiverAddress),
+      ])
+
+      return this.providers.http.sendTx({ sendTxBody: { hex } })
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async getFeeData(
+    input: GetFeeDataInput<KnownChainIds.SolanaMainnet>,
+  ): Promise<FeeDataEstimate<KnownChainIds.SolanaMainnet>> {
+    const { baseFee, fast, average, slow } = await this.providers.http.getPriorityFees()
+
+    const computeUnits = await this.providers.http.estimateFees({
+      estimateFeesBody: { message: input.chainSpecific?.message },
+    })
+
+    return {
+      fast: {
+        txFee: bn(microLamportsToLamports(fast)).times(computeUnits).plus(baseFee).toFixed(),
```

Comment:
I don't really get why we need `microLamportsToLamports`, getPriorityFees doesn't seems like returning micro-lamports in my memory?

Reply:
![image](https://github.com/user-attachments/assets/ba0f5e23-b630-429f-bbbd-b37d45766568)

it is micro-lamports per compute unit, but micro-lamports are not really a thing we care about in the UI, so converting to base unit (lamports) for the displayed tx fee

Reply:
Ok, weird but makes sense from their doc! lgtm then